frontend/src/data/api-service.ts

```diff
@@ -90,10 +92,15 @@ export const createApiService = ({
   const axiosParams: AxiosRequestConfig = {
     baseURL: isVersioned ? `${baseUrl}v1/` : baseUrl,
     timeout: DEFAULT_REQUEST_TIMEOUT,
+    headers: { "User-Agent": userAgent },
   }
+
   if (accessToken) {
-    axiosParams.headers = { Authorization: `Bearer ${accessToken}` }
+    if (axiosParams.headers) {
```

Comment:
Why the `if`? Isn't `headers` hardcoded to be present on L95?

Reply:
The if is pretty confusing. When I first read it, it made me think that somehow `headers` could be undefined, even though it's _always_ on the axios params above.

There are two ways of doing this. One is purely in the type space and the other would be a slight change in the code to remove the if:

1. Change `const axiosParams: AxiosRequestConfig` to `const axiosParams: Required<Pick<AxiosRequestConfig, "headers">> & AxiosRequestConfig`. That tells TypeScript to remove `undefined` from the `headers` property, removing the need to check the `if` below. It's a bit wordy, but removes the need for unnecessary code. It also doesn't introduce the risk of a cast because if `headers` is _not_ defined on the right-hand side of the assignment expression, the type check will fail because the object won't fulfil the requirements of the type (`headers` being a headers object).
2. Cast headers below in the assignment to remove undefined: ``(axiosParams.headers as Exclude<typeof axiosParams["headers"], undefined>)["Authorization"] = `Bearer ${accessToken}` ``. To me this is objectively worse because if `headers` was removed from the `axiosParams` assignment above, there would be no type error.

I have a strong preference for the first option. To reduce the wordiness of it, you could create a new type:

```ts
/**
 * Openverse Axios request config with adjusted types for our use-case.
 */
type OpenverseAxiosRequestConfig = Required<Pick<AxiosRequestConfig, "headers">> & AxiosRequestConfig
```

Then use `OpenverseAxiosRequestConfig` in place of `AxiosRequestConfig` in the `axiosParams` assignment above.